### PR TITLE
Navigation: next button behaviour

### PR DESCRIPTION
**Pull request by Itxaka**
- Use the step factory to track and set the step status to finished
- Link the steps directive to enable/disable the next button based on the step status
- Set stepCompleted in all the individual steps on completition

### CODE REVIEW

**Comment by Itxaka** [comment]
[(3) S49P20 #6 Layout - Navigation bar](https://trello.com/c/LNr710jO/104-3-s49p20-6-layout-navigation-bar)

File: assets/app/features/upgrade/upgrade.controller.js [hunk 1]
```diff
@@ -18,7 +18,8 @@
         vm.steps = {
             list: [],
             nextStep: nextStep,
-            isLastStep: isLastStep
+            isLastStep: isLastStep,
+            stepFinished: upgradeStepsFactory.stepFinished
```
Comment (skazi0) [inline assets/app/features/upgrade/upgrade.controller.js]: shouldn't it be `isStepFinished()`? (or whatever comes out of previous comment)

File: assets/app/features/upgrade/steps.factory.js [hunk 2]
```diff
@@ -10,11 +10,20 @@
         var factory = {
             steps: initialSteps(),
             activeStep: {},
-            refeshStepsList: refeshStepsList
+            refeshStepsList: refeshStepsList,
+            setCurrentStepCompleted: setCurrentStepCompleted,
+            isStepFinished: isStepFinished
```
Comment (skazi0) [inline assets/app/features/upgrade/steps.factory.js]: Maybe change this to `isCurrentStepCompleted` for consistency with `setCurrentStepCompleted`?

**Review by skazi0 — CHANGES_REQUESTED** [review]

File: assets/app/features/upgrade/upgrade.controller.js (hunk 1 above)
Comment (Itxaka) [reply assets/app/features/upgrade/upgrade.controller.js]: THANK YOU!

I knew it was missing somewhere and could not find it for the life of me.

File: assets/app/features/upgrade/steps.factory.js (hunk 2 above)
Comment (Itxaka) [reply assets/app/features/upgrade/steps.factory.js]: Good point, will change it!

**Review by skazi0 — APPROVED** [review]

**Pull request merged**